tokio-test/tests/io.rs

```diff
@@ -84,3 +84,17 @@ async fn mock_panics_write_data_left() {
     use tokio_test::io::Builder;
     Builder::new().write(b"write").build();
 }
+
+#[tokio::test]
+async fn write_with_failing_assert_callback() {
+    static mut CALLED: bool = false;
+
+    let mut mock = Builder::new()
+        .write(b"test")
+        .set_assert_callback(|_, _| unsafe { CALLED = true })
```

Comment:
a lock is probably better here?

Reply:
My intention was to have closure type `Fn` versus `FnMut`. Mutating between unit threads seems unnecessary. Is there a lock I can use that will work in an `Fn` closure?

Reply:
All locks work in `Fn` closures — that's the point of locks. A `Mutex` or `AtomicBool` would have worked.

Reply:
Yup, `std::sync::Mutex` works

Reply:
Ah ok! Thanks @Darksonn . I had tried `AtomicBool` originally and ran into `Fn` vs `FnMut` issues but `Mutex` works fine. Thanks for the clarification.